File: Models/Platform.cs
using System;
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;

namespace AccountAPI.Models
{
	public class Platform
    {
        [Key]
        public int PlatformId {get;set;}
        [Required]
        public string Name {get;set;}
        public string URLToDocumentation {get;set;}
        public DateTime CreatedAt {get;set;} = DateTime.Now;
        public DateTime UpdatedAt {get;set;} = DateTime.Now;

        public ICollection<Game> Games {get;} = new List<Game>();
        public ICollection<Account> Accounts {get;} = new List<Account>();
    }
}

File: Client/src/app/Models/platform.ts
export interface PlatformList {
    Id: number;
    Platform: string;
    GameCount: number;
}

export interface PlatformAdd {
    Name: string;
}

export interface PlatformEdit {
    PlatformId: number;
    Name: string;
    URLToDocumentation: string;
}

export interface Platform {
    PlatformId: number;
    Name: string;
    URLToDocumentation: string;
    CreatedAt: string;
    UpdatedAt: string;
    Games: PlatformGames[];
    Accounts: PlatformAccounts[];
}

export interface PlatformGames {
    Id: number;
    Name: string;
}

export interface PlatformAccounts {
    Id: number;
    Username: string;
}

export interface Platforms {
    Id: number;
    Platform: string;
}
  